python-packaging/execute-code.md

```diff
@@ -0,0 +1,167 @@
+Windows to associate the file with Python, as opposed to on Linux where `.py` is a convention and the shebang associates
+the file with Python.
+
+Additionally, most Windows Python installs come with the [Python Launcher](https://docs.python.org/3/using/windows.html#python-launcher-for-windows)
+which, in addition to allowing specifying the version of Python, can also read shebang lines and emulate some of that behavior.
+This allows for shebang lines to be re-used between Linux, macOS, and Windows systems. However, on Windows the command must still
+be prefaced with another command (`py`).
+
+```bash
+py my_program.py
+# ✨ Hello from Python ✨
+```
+
+:::{tip}
+While there is no in-source format that can tell Windows what to do with a Python code file, executing a
+Python file with a shebang on Windows also does not cause any issues. Python just sees the whole line as
+a comment and ignores it! So even if you develop on Windows, it may be a good idea to add the shebang as
+the first line of your scripts, so that colleagues on different systems can also run it.
+:::
+
+## Executable comparisons
+
+### Pros of passing a file to `python`:
+- don't need execute permissions
+- works for every system
+- explicit about what you expect to happen
+
+### Pros of inserting a shebang to the file:
+- file is associated with specific python
+  - don't have to remember which
+- don't have to use the `python` command
+  - don't have to even remember it is a Python script
+
+
+# Share Code
+
+## Execute a python package
+
+In [Code Workflow Logic][Code Workflow Logic] you learned of the two primary ways to execute a stand-alone Python script.
+There are two other ways to execute Python as commands, both of which work for code that has been formatted as a package.
+
+### Entrypoints
+
+There is a special `entrypoint` a package can specify in its configuration which will direct installers to create an
+executable command. Entrypoints are a general purpose plug-in system for Python packages, but the
+[`console_scripts`](https://packaging.python.org/en/latest/specifications/entry-points/#use-for-scripts)
+entry is specifically targeted at creating executable commands on systems that install the package.
+
+The target of a `scripts` definition should be one function within your package, which will be directly executed
+when the command is invoked in the shell. A `scripts` definition in your `pyproject.toml` looks like:
+
+```toml
+[project.scripts]
+COMMAND = "my_package.my_module:my_function"
+```
+
+where `COMMAND` is the name of the command that will be made available after installation, `my_package` is the name of
+your top-level package import, `my_module` is the name of any sub-modules in your package (optional, or may be
+repeated as necessary to access the correct sub-module), and `my_funciton` is the function that will be called
+(without parameters) when the command is invoked.
```

Comment:
i think it would be worth it to make a fake package tree diagram with a quick little demo of what `my_module` looks like with `my_function` inside of it here, again just making things concrete for learners :) - you say all that in text here but pictures are nice to be able to point to.